sound/soc/sof/ipc4-topology.c

```diff
@@ -1014,20 +983,21 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 	}
 
 	/* copy input format */
-	memcpy(&base_config->audio_fmt, &available_fmt->sink_audio_fmts[i], sizeof(*fmt));
+	memcpy(&base_config->audio_fmt, &available_fmt->sink_audio_fmts[i].audio_fmt,
+	       sizeof(*out_format));
 
 	/* set base_cfg ibs/obs */
-	base_config->ibs = available_fmt->base_config[i].ibs;
-	base_config->obs = available_fmt->base_config[i].obs;
+	base_config->ibs = available_fmt->sink_audio_fmts[i].buffer_size;
+	base_config->obs = available_fmt->out_audio_fmt[i].buffer_size;
```

Comment:
sink_audio -> source_audio. Le'ts not mix sink and out, please,

Reply:
All fixed by the end of the series too